app/Http/Controllers/Collectible/CollectibleController.php

```diff
@@ -0,0 +1,397 @@
+            'collectibles'          => $collectibles,
+        ]);
+    }
+
+    /**
+     * Display My Collectibles.
+     */
+    public function myCollectibles(): \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+    {
+        $user = auth()->user();
+
+        $collectibleCategories = CollectibleCategory::get();
+        $collectibles = Collectible::whereHas('items', function ($query) use ($user): void {
+            $query->where('user_id', '=', $user->id);
+        })
+            ->with('category')
+            ->get();
+        $collectiblesByCategory = $collectibles->groupBy('category_id');
+
+        return view('collectible.my_collectible', [
+            'collectibleCategories'  => $collectibleCategories,
+            'collectiblesByCategory' => $collectiblesByCategory,
+        ]);
+    }
+
+    /**
+     * Display A Collectible.
+     */
+    public function show(Collectible $collectible): \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+    {
+        $user = User::withSum('seedingTorrents as seedsize', 'size')
+            ->withAvg('history', 'seedtime')
+            ->findOrFail(auth()->user()->id);
+
+        $collectible->load('requirements');
+        $collectibleItems = CollectibleItem::where('collectible_id', $collectible->id);
+        $userOwns = $collectibleItems->where('user_id', $user->id)->exists();
+        $transactions = CollectibleTransaction::where('collectible_id', $collectible->id)
+            ->latest()
+            ->take(20)
+            ->get();
+        $offers = CollectibleOffer::where('collectible_id', $collectible->id)
+            ->whereNull('filled_when')
+            ->get();
+        $userIsSelling = CollectibleOffer::where('collectible_id', $collectible->id)
+            ->where('user_id', $user->id)
+            ->exists();
+        $userTransaction = CollectibleTransaction::where('collectible_id', $collectible->id)
+            ->where('buyer_id', $user->id)
+            ->latest()
+            ->first();
+        $availableCount = CollectibleItem::where('collectible_id', $collectible->id)->whereNull('user_id')->count();
+
+        // Evaluate the requirements for this collectible for the current user
+        $userAttributes = [
+            'uploaded' => [
+                'value' => $user->uploaded ?? 0,
+                'label' => 'Account Upload'
+            ],
+            'seedsize' => [
+                'value' => $user->seedsize ?? 0,
+                'label' => 'Seeding Size'
+            ],
+            'average_seedtime' => [
+                'value' => $user->history_avg_seedtime ?? 0,
+                'label' => 'Average Seedtime'
+            ],
+            'ratio' => [
+                'value' => $user->ratio ?? 0,
+                'label' => 'Account Ratio'
+            ],
+            'age' => [
+                'value' => $user->created_at->diffInSeconds(now()),
+                'label' => 'Account Age'
+            ],
+        ];
```

Comment:
This should probably be in the blade instead

Reply:
Does this really need to be moved to the blade? IMO, it should be calculated in the controller. Changing it would also mean to move the whole logic from https://github.com/HDInnovations/UNIT3D/blob/af3c5545413e845eac8af1fe3fca8c7487cbd503/app/Http/Controllers/Collectible/CollectibleController.php#L95-L116 to the view too(?).